poetry/repositories/legacy_repository.py

```diff
@@ -304,6 +304,13 @@ def package(self, name, version, extras=None):  # type: (...) -> Package
 
             return package
 
+    def find_links_for_package(self, package):
```

Comment:
minor, but you missed the type annotation here :) have you considered turning on `disallow_untyped_defs`?